dpnp/backend/kernels/dpnp_krnl_common.cpp

```diff
@@ -111,8 +111,16 @@ cl::sycl::event dot(cl::sycl::queue& queue,
                                                  std::plus<_DataType_output>(),
                                                  cl::sycl::property::reduction::initialize_to_identity{}),
                              [=](cl::sycl::id<1> idx, auto& sum) {
-                                 sum += static_cast<_DataType_output>(input1_in[idx * input1_strides]) *
-                                        static_cast<_DataType_output>(input2_in[idx * input2_strides]);
+                                 if (input1_strides > 0)
```

Comment:
What if `input1_strides = 0`?
Will it work if `input1_strides > 0` and `input2_strides < 0` or vise versa?